### PR TITLE
Automatically size table based on Terminal width

### PR DESCRIPTION
**Pull request by Piccirello**
This change adapts the CLI's output to the current terminal size. It results in much better looking output at smaller sizes.

Closes ENG-2602

### CODE REVIEW

File: pkg/printer/print.go [hunk 1]
```diff
@@ -64,10 +75,14 @@ func Table(headers []string, rows [][]string, options tableOptions) {
 	}
 	t.AppendHeader(tableHeaders)
 
+	numCols := numColumns(rows)
+	maxColWidths := maxColWidths(rows, numCols)
+	colWidths := optimalColWidths(maxColWidths, numCols)
```
Comment (Piccirello) [inline pkg/printer/print.go:76]: These functions might be worth unit testing

File: pkg/printer/enclave.go [hunk 2]
```diff
@@ -0,0 +1,363 @@
+/*
```
Comment (Piccirello) [inline pkg/printer/enclave.go:1]: The commit `chore: move enclave print functions to separate file` is purely moving functions- there aren't any logic changes.

**Review by nmanoogian — APPROVED** [review]
This is awesome!

**Review by rgmvisser — APPROVED** [review]

**Comment by rgmvisser** [comment]
Nice papercut!

**Review by apazzolini — APPROVED** [review]
Works well in tmux, nice one!

**Comment by Piccirello** [comment]
> Works well in tmux, nice one!

Thanks for testing that use case!

**Pull request merged**